index.js

```diff
@@ -110,8 +110,9 @@ Prober.prototype.probe = function probe(request, bypass, callback) {
     var wrappedCallback;
     if (this.detectFailuresByCallback) {
         wrappedCallback = function(err, resp) {
-            var errResponse = resp && !isNaN(resp.statusCode) &&
-                resp.statusCode >= 500;
+            var errResponse = resp &&
+                (!isNaN(resp.statusCode) && resp.statusCode >= 500 || // HTTP
+                resp.hasOwnProperty('ok') && !resp.ok); // tchannel
```

Comment:
This kind of hack is bad :(.

Best to add a new `Prober.prototype.tchannelProbe()` method that dispatches to `customProbe()`